pkg/build/builder/docker.go

```diff
@@ -70,40 +66,27 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	glog.V(4).Infof("Starting Docker build from %s/%s BuildConfig ...", d.build.Namespace, d.build.Name)
-	var push bool
-
-	// if there is no output target, set one up so the docker build logic
-	// will still work, but we won't push it at the end.
-	if d.build.Spec.Output.To == nil || len(d.build.Spec.Output.To.Name) == 0 {
```

Comment:
In general, I don't like seeing changes like this coupled with significant test changes... it makes it really hard to tell if previous API behavior still works. Can we add additional testcases while keeping the existing ones as-is?